org-roam.el

```diff
@@ -648,7 +648,8 @@ This needs to be quick/infrequent, because this is run at
     (when (and (or redisplay
                    (not (eq org-roam--current-buffer buffer)))
                (eq 'visible (org-roam--current-visibility))
-               (buffer-local-value 'buffer-file-truename buffer))
+               (buffer-local-value 'buffer-file-truename buffer)
+               (org-roam-cache-initialized))
```

Comment:
move this check up? it might be cheaper.

Reply:
I'm not sure if it is, but I'm made the switch.  I'll think about how to optimize the cache checks better.  There should be a better way than doing a string comparison.  Maybe the lookup is done once per buffer and then the cache object is made buffer local?  I'd have to not replace the cache object but rather update them.